scipy/special/_basic.py

```diff
@@ -2639,22 +2637,20 @@ def comb(N, k, exact=False, repetition=False, legacy=True):
     220
 
     """
+    if legacy:
+        warnings.warn(
+            "Using 'legacy=True' is deprecated and will raise an error by "
+            "Scipy 1.13.0. If you want to keep the legacy behaviour, cast "
```

Comment:
We're only showing this warning if someone sets `legacy=True`, but if someone has added `legacy=False`, they will not see the warning and - AFAIU - will be broken by the removal in 1.13. I think the warning here is fine, but we should not error on `legacy=False` in 1.13

Reply:
Thanks for the review @h-vetinari. Do you seen any reason we can't set the default to `None` for this release then have seperate warnings for `legacy=True` and `legacy=False` so that we can remove `legacy` by 1.13?

Reply:
It's possible. My point in #15722 was that we shouldn't punish those most diligent to respond to the original deprecation warning (by adding `legacy=False`) by showing a warning again right away. Bug I guess it's a fairly niche problem and people will just move on.